dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientChannels.java

```diff
@@ -97,4 +128,64 @@ private static URL url(String uri) {
             throw new SafeIllegalArgumentException("Failed to parse URL", e);
         }
     }
+
+    private enum NullCredentialsProvider implements CredentialsProvider {
```

Comment:
I want to avoid dealing with credentials in all possible cases so we don't have to deal with folks getting used to client-specific behavior.
When proxy credentials are provided I want to make sure they can only be used by proxies by disabling the target auth impl, and all schemes except basic to match okhttp.